daemon/src/maker_cfd.rs

```diff
@@ -205,18 +167,15 @@ impl<O, M, T, W> Actor<O, M, T, W> {
 
     async fn handle_inc_roll_over_protocol_msg(
         &mut self,
-        taker_id: Identity,
+        order_id: OrderId,
         msg: wire::RollOverMsg,
     ) -> Result<()> {
-        match &mut self.roll_over_state {
-            RollOverState::Active { taker, sender, .. } if taker_id == *taker => {
-                sender.send(msg).await?;
-            }
-            RollOverState::Active { taker, .. } => {
-                anyhow::bail!("Currently rolling over with different taker {}", taker)
-            }
-            RollOverState::None => {}
-        }
+        let rollover_actor = self
+            .rollover_actors
+            .get_connected(&order_id)
+            .context("rollover actor is not connected")?;
+
+        rollover_actor.send(msg).await?;
 
         Ok(())
     }
```

Comment:
This never happens because you are doing the routing inside the `maker_inc_connections::Actor`.